tests/rptest/tests/alter_topic_configuration_test.py

```diff
@@ -132,6 +133,53 @@ def test_configuration_properties_name_validation(self):
         # topic spec shouldn't change
         assert new_spec == spec
 
+    @cluster(num_nodes=3)
+    def test_segment_size_validation(self):
+        topic = self.topics[0].name
+        kafka_tools = KafkaCliTools(self.redpanda)
+        initial_spec = kafka_tools.describe_topic(topic)
+        self.redpanda.set_cluster_config({"log_segment_size_min": 1024})
+        try:
+            self.client().alter_topic_configs(
+                topic, {TopicSpec.PROPERTY_SEGMENT_SIZE: 16})
+        except subprocess.CalledProcessError as e:
+            assert "is outside of allowed range" in e.output
+
+        assert initial_spec.segment_bytes == kafka_tools.describe_topic(
+            topic
+        ).segment_bytes, "segment.bytes shouldn't be changed to invalid value"
+
+        # change min segment bytes redpanda property
+        self.redpanda.set_cluster_config({"log_segment_size_min": 1024 * 1024})
+        # try setting value that is smaller than requested min
+        try:
+            self.client().alter_topic_configs(
+                topic, {TopicSpec.PROPERTY_SEGMENT_SIZE: 1024 * 1024 - 1})
+        except subprocess.CalledProcessError as e:
+            assert "is outside of allowed range" in e.output
+
+        assert initial_spec.segment_bytes == kafka_tools.describe_topic(
+            topic
+        ).segment_bytes, "segment.bytes shouldn't be changed to invalid value"
+        valid_segment_size = 1024 * 1024 + 1
+        self.client().alter_topic_configs(
+            topic, {TopicSpec.PROPERTY_SEGMENT_SIZE: valid_segment_size})
+        assert kafka_tools.describe_topic(
+            topic).segment_bytes == valid_segment_size
+
+        self.redpanda.set_cluster_config(
+            {"log_segment_size_max": 10 * 1024 * 1024})
+        # try to set value greater than max allowed segment size
+        try:
```

Comment:
nit: there's an `expect_exception` helper that can make cases like this a bit more explicit (and it will throw if the expected exception was not raised)